### PR TITLE
Closes #263 (draft PR)

### LINKED ISSUE #263: Rules methods cleanup: `NextBestNCRM` signature
Please edit appropriately below if needed.

To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] Remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] For classes:
    - [x] Test default constructor
    - [x] For reference classes: test methods
  - [x] For functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] Keep externally used names so that we don't break downstream code
  - [x] Remove `keywords` tags
  - [x] For functions or methods:
    - [x] Add assertions for arguments using `checkmate` functions
    - [x] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [x] Consider renaming internal variables etc. if needed
    - [x] Avoid explicit use of `return`
  - [x] For classes:
    - [x] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [x] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #263 (draft PR)

### CODE REVIEW

**Comment by wwojciech** [comment]
Hi @danielinteractive I am submitting this PR as draft, but it is actually fully ready from my side. Just wanted to check with you what do you think on this. Tx!

File: NEWS.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Version 1.0.9000.9040
+* Warning message not printed by `nextBest` methods if finite `doselimit` not specified.
```
Comment (danielinteractive) [inline NEWS.md:2]: can you reword this? not sure I understand it.

File: R/Rules-class.R [hunk 2]
```diff
@@ -95,10 +95,11 @@ NextBestMTD <- function(target, derive) {
 #' Otherwise, the highest admissible dose is the next recommended dose.
 #'
 #' @slot target (`numeric`)\cr the target toxicity interval (limits included).
+#'   It is used to filter probability samples.
```
Comment (danielinteractive) [inline R/Rules-class.R:98]: this we don't need. it is used to calculate target probabilities. the filtering you mention and we discussed yesterday is a technical implementation detail not relevant here

File: R/Rules-class.R [hunk 3]
```diff
@@ -95,10 +95,11 @@ NextBestMTD <- function(target, derive) {
...
 #' @slot overdose (`numeric`)\cr the overdose toxicity interval (lower limit
-#'   excluded, upper limit included).
-#' @slot max_overdose_prob (`proportion`)\cr maximum overdose probability that
-#'   is allowed.
+#'   excluded, upper limit included). It is used to filter probability samples.
```
Comment (danielinteractive) [inline R/Rules-class.R:100]: also here same comment

File: R/Rules-class.R [hunk 4]
```diff
@@ -95,10 +95,11 @@ NextBestMTD <- function(target, derive) {
...
+#' @slot max_overdose_prob (`proportion`)\cr maximum overdose posterior
+#'   probability that is allowed.
```
Comment (danielinteractive) [inline R/Rules-class.R:102]: that is good.

File: R/Rules-methods.R [hunk 5]
```diff
@@ -75,21 +75,18 @@ setMethod(
                         model,
                         data,
                         ...) {
+    # TODO: Below line - to be removed after issue #261 is solved.
```
Comment (danielinteractive) [inline R/Rules-methods.R:78]: please no TODOs in code. issue is sufficient.

File: R/Rules-methods.R [hunk 6]
```diff
@@ -139,216 +136,168 @@ setMethod(
...
-                           aes(x=x, y=y),
-                           stat="identity",
-                           position="identity",
-                           width=min(diff(data@doseGrid)) / 2,
-                           colour="darkgreen",
-                           fill="darkgreen") +
-                               xlab("Dose") +
-                                   ylab(paste("Target probability [%]")) +
-                                       ylim(c(0, 100))
+    # TODO: Below line - to be removed after issue #261 is solved.
```
Comment (danielinteractive) [inline R/Rules-methods.R:165]: same here

**Review by danielinteractive — COMMENTED** [review]
thanks Wojciech, good work, see initial comments

**Comment by wwojciech** [comment]
Hi @dinakar29 , I converted this draft PR into a regular PR, but the automation checks do not want to start.... Can you please have a look at this? Tx a lot!

**Comment by danielinteractive** [comment]
Thanks @wwojciech , can you remove this filtering wording? It is confusing. Basically we don't need to say anything around filtering of probability samples.
The logic is: the specified intervals are used to calculate posterior probabilities for target dosing and overdosing. Then, the posterior probability for overdosing together with the corresponding threshold is used to filter out too toxic, i.e. overdosing, doses.

**Comment by wwojciech** [comment]
> Thanks @wwojciech , can you remove this filtering wording? It is confusing. Basically we don't need to say anything around filtering of probability samples. The logic is: the specified intervals are used to calculate posterior probabilities for target dosing and overdosing. Then, the posterior probability for overdosing together with the corresponding threshold is used to filter out too toxic, i.e. overdosing, doses.

Sure, but actually you wrote in your review that `this is ok`.

**Comment by danielinteractive** [comment]
@wwojciech yeah that refers to that single line only :-)
I was just now referring to this to remove:

**Comment by wwojciech** [comment]
> @wwojciech yeah that refers to that single line only :-) I was just now referring to this to remove: image

Sure, no problem :) I removed it all now :)

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**